// @twa-dev/mark42/dist/Components/ImageAvatar/ImageAvatar.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["src", "alt", "size"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

import { memo } from "react";
import styles from "./ImageAvatar.styles";
import { Avatar } from "../Avatar";
import { Image } from "../Image";
import { jsx as _jsx } from "react/jsx-runtime";
export var ImageAvatar = /*#__PURE__*/memo(function (_ref) {
  var src = _ref.src,
      alt = _ref.alt,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 40 : _ref$size,
      restProps = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_jsx(Avatar, _objectSpread(_objectSpread({
    size: size
  }, restProps), {}, {
    children: /*#__PURE__*/_jsx(Image, {
      src: src,
      alt: alt,
      className: styles.img
    })
  }));
});
ImageAvatar.defaultProps = {
  size: 40
};
ImageAvatar.displayName = "ImageAvatar";
